### PR TITLE
Metrics for witness start and self-update

### PR DESCRIPTION
**Pull request by mhutchinson**

### CODE REVIEW

File: trusted_applet/main.go [hunk 1]
```diff
@@ -78,6 +79,20 @@ var (
...
+var (
+	doOnce                       sync.Once
+	counterWitnessStarted        monitoring.Counter
+	counterFirmwareUpdateAttempt monitoring.Counter
+)
+
+func initMetrics() {
+	doOnce.Do(func() {
+		mf := monitoring.GetMetricFactory()
+		counterWitnessStarted = mf.NewCounter("witness_started", "Number of times the witness was started")
```
Comment (AlCutter) [inline trusted_applet/main.go:91]: Should we call this `witness_restarts`? I think it's mainly going to count the number of restarts due to DHCP lease expiries, if the device crashes and restarts I suspect this'll continue to show the same number to prometheus and it won't notice, unless there's some clever trickery behind the scenes?

File: trusted_applet/main.go [hunk 2]
```diff
@@ -78,6 +79,20 @@ var (
...
+		counterFirmwareUpdateAttempt = mf.NewCounter("firmware_update_attempt", "Number of times the updater ran to check if firmware could be updated")
```
Comment (AlCutter) [inline trusted_applet/main.go:92]: How about `num_firmware_update_checks`? Then we probably want a `num_firmware_update_attempts` success/fail to detect problems?

**Review by AlCutter — APPROVED** [review]

File: trusted_applet/main.go (hunk 1 above)
Comment (mhutchinson) [reply trusted_applet/main.go:92]: As discussed we'll keep this name as it is.

File: trusted_applet/main.go (hunk 2 above)
Comment (mhutchinson) [reply trusted_applet/main.go:93]: Added a success counter.

**Pull request merged**